metadata-ingestion/src/datahub/ingestion/source/iceberg/__init__.py

```diff
@@ -0,0 +1,341 @@
+        namespaces = self.fsClient.get_paths(path=self.config.base_path, recursive=False)
+        namespace: PathProperties
+        for namespace in namespaces:
+            namespaceTables = self.fsClient.get_paths(path=namespace.name, recursive=False)
+            tablePath : PathProperties
+            for tablePath in namespaceTables:
+                try:
+                    # TODO Stripping 'base_path/' from tablePath.  Weak code, need to be improved later
+                    namespace, tableName = tablePath.name[len(self.config.base_path)+1:].split('/')
+                    # TODO I'm replacing '/' with '.' for the dataset name.  Is this wrong?  Should I keep the slashes?
+                    datasetName = f"{namespace}.{tableName}"
+                    if not self.config.table_pattern.allowed(datasetName):
+                        self.report.report_dropped(datasetName)
+                        continue
+                    else:
+                        table: Table = self.icebergClient.load(f"{self.config.filesystem_url}/{tablePath.name}")
+                        yield from self._createIcebergWorkunit(datasetName, table)
+                except Exception as e:
+                    self.report.report_failure("general", f"Failed to create workunit: {e}")
+                    LOGGER.exception(f"Exception while processing table {self.config.filesystem_url}/{tablePath.name}, skipping it.", e)
+
+    def _createIcebergWorkunit(self, datasetName: str, table: Table) -> Iterable[MetadataWorkUnit]:
+        self.report.report_table_scanned(datasetName)
+        datasetUrn = make_dataset_urn(self.platform, datasetName, self.config.env)
+        dataset_snapshot = DatasetSnapshot(
+            urn=datasetUrn,
+            aspects=[],
+        )
+
+        customProperties = dict(table.properties())
+        customProperties["location"] = table.location()
+        customProperties["snapshot-id"] = str(table.current_snapshot().snapshot_id)
+        customProperties["manifest-list"] = table.current_snapshot().manifest_location
+        dataset_properties = DatasetPropertiesClass(
+            tags=[],
+            customProperties=customProperties,
+        )
+        dataset_snapshot.aspects.append(dataset_properties)
+
+        # TODO: The following seems to go in OperationClass.  It is usually part of the "usage" source.  Should I create a new python module?
+        last_updated_time_millis=table.current_snapshot().timestamp_millis # Should we use last-updated-ms instead?  YES
+        operation_aspect = OperationClass(
+            timestampMillis=get_sys_time(),
+            lastUpdatedTimestamp=last_updated_time_millis,
+            operationType=OperationTypeClass.UNKNOWN,
+        )
+        # TODO: How do I add an operation_aspect?
+        
+        if "owner" in table.properties():
+            ownerEmail = table.properties()["owner"]
+            owners = [OwnerClass(
+                owner=f"urn:li:corpuser:{ownerEmail}",
```

Comment:
For readability, could you use `make_user_urn` from `mce_builder` here? https://github.com/linkedin/datahub/blob/master/metadata-ingestion/src/datahub/emitter/mce_builder.py#L85